pkg/synopsysctl/cmd_create.go

```diff
@@ -161,96 +242,132 @@ var createBlackDuckCmd = &cobra.Command{
 			cmd.Help()
 			return fmt.Errorf("this command takes 1 argument")
 		}
-		// Check the user's flags
-		err := createBlackDuckCtl.CheckSpecFlags(cmd.Flags())
+		return nil
+	},
+	PreRunE: createBlackDuckPreRun,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		mockMode := cmd.Flags().Lookup("mock").Changed
+		blackDuckName, blackDuckNamespace, scope, err := getInstanceInfo(false, args[0], util.BlackDuckCRDName, "", namespace)
 		if err != nil {
-			cmd.Help()
 			return err
 		}
-		// Set the base spec
-		if !cmd.Flags().Lookup("template").Changed {
-			baseBlackDuckSpec = defaultBaseBlackDuckSpec
+		blackDuck := updateBlackDuckSpecWithFlags(cmd, blackDuckName, blackDuckNamespace)
+
+		// If mock mode, return and don't create resources
+		if mockMode {
+			log.Debugf("generating CRD for Black Duck '%s' in namespace '%s'...", blackDuckName, blackDuckNamespace)
+			return PrintResource(*blackDuck, mockFormat, false)
+		}
+
+		log.Infof("creating Black Duck '%s' in namespace '%s'...", blackDuckName, blackDuckNamespace)
+		// Check if Synopsys Operator is running
+		if err := checkOperatorIsRunning(scope, blackDuckNamespace); err != nil {
+			return err
+		}
+		// Verifying only one Black Duck instance per namespace
+		blackducks, err := util.ListHubs(blackDuckClient, blackDuckNamespace)
+		if err != nil {
+			return fmt.Errorf("unable to list Black Duck instances in namespace '%s' due to %+v", blackDuckNamespace, err)
+		}
+		for _, v := range blackducks.Items {
+			if strings.EqualFold(v.Spec.Namespace, blackDuckNamespace) {
+				return fmt.Errorf("due to issues with this version of Black Duck, only one instance per namespace is allowed")
+			}
+		}
+		// Create namespace for the Black Duck instance
```

Comment:
@rrati , @msenmurugan and I think you need to create the namespace first because `blackDuckClient.SynopsysV1().Blackducks(blackDuckNamespace).Create(blackDuck)` takes in the `blackDuckNamespace` when deploying. I think that making the namespace not required could be an issue we look into for a future release.

Reply:
The namespace should either already exist because the operator is running in that NS (cluster mode) or will be created/used by the operator (in cluster mode).  When would we have a case of creating a CRD and the operator wouldn't exist in the NS or be responsible for creating the NS?